### PR TITLE
[CWORKS-1117] Add dyn_node resource

### PR DESCRIPTION
**Pull request by wershlak**
Ready to review and merge. See comments

### CODE REVIEW

File: dyn/resource_dyn_node.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+			},
+
+		},
+	}
+}
+
+func expandSet(recs []interface{}) []string {
+	records := make([]string, 0, len(recs))
+	for _, r := range recs {
+		s := r.(string)
```
Comment (imjoshholloway) [inline dyn/resource_dyn_node.go]: Probably want to add a check here:
```go
s, ok := r.(string)
if !ok {
    // item isn't a string so can't add it to the slice
    continue
}
```

Comment (wershlak) [reply dyn/resource_dyn_node.go]: Cool. Not sure that can happen here but still better to add it.

**Comment by wershlak** [comment]
Updated the code.
I'd really like to redo this someday and model the whole node but didn't think about it before I started.

Now you can have multiple record types as different resources like:
```
resource "dyn_node" "test-jeff6" {
  type = "A"
  zone = "splunkworks.lol"
  values = ["192.168.1.1", "192.168.1.2", "192.168.1.3", "192.168.1.4", "192.168.1.5", "192.168.1.6", "192.168.1.7", "192.168.1.8", "192.168.1.9", "192.168.1.10", "192.168.1.11", "192.168.1.12", "192.168.1.13"]
  name = "jeffdyntest6"
}

resource "dyn_node" "test-jeff6-MX" {
  type = "MX"
  zone = "splunkworks.lol"
  values = ["10 mail.x.com", "10 mail1.x.com", "10 mail3.x.com"]
  name = "jeffdyntest6"
}
```
But it should really be like:

```
resource "dyn_node" "test-jeff6" {
  zone = "splunkworks.lol"
  name = "jeffdyntest6"

  records {
    type = "A"
    values = ["192.168.1.1", "192.168.1.2", "192.168.1.3", "192.168.1.4", "192.168.1.5", "192.168.1.6", "192.168.1.7", "192.168.1.8", "192.168.1.9", "192.168.1.10", "192.168.1.11", "192.168.1.12", "192.168.1.13"]
  }
  
  records {
    type = "MX"
    values = ["10 mail.x.com", "10 mail1.x.com", "10 mail3.x.com"]
  }
  
}
```

Anyway, this seems solid for our use case. See testing in the TF PR

https://github.com/SplunkCloud/terraform-stack-infrastructure/pull/154

File: vendor/vendor.json [hunk 2]
```diff
@@ -556,10 +556,12 @@
...
 		},
 		{
-			"checksumSHA1": "cAfESkU79dp//i0lBzwS+BLPxU8=",
+			"checksumSHA1": "9u8I/nG8pMI5O6PTp/bjG5gJdOk=",
 			"path": "github.com/splunkcloud/go-dynect/dynect",
-			"revision": "0313d67515d79eb8c494651540917310dfbdf12e",
-			"revisionTime": "2018-06-15T18:20:19Z"
+			"revision": "3de358b54fa77adb1215f5de0638f2d9ad293b29",
+			"revisionTime": "2018-08-06T22:53:28Z",
+			"version": "cworks_1117_add_node",
```
Comment (imjoshholloway) [inline vendor/vendor.json]: We keeping this as is or do we want to merge the client changes and then update this?

Comment (wershlak) [reply vendor/vendor.json]: Yeah I'll open a PR over there.

**Pull request merged**